storage/innobase/include/ut0counter.h

```diff
@@ -114,6 +115,12 @@ struct ib_counter_t {
 	@see srv_stats */
 	struct ib_counter_element_t {
 		MY_ALIGNED(CACHE_LINE_SIZE) std::atomic<Type> value;
+		ib_counter_element_t& operator =(
+			const ib_counter_element_t& from) {
+			value.store(from.value.load(std::memory_order_relaxed),
+				std::memory_order_relaxed);
+			return *this;
+		}
 	};
```

Comment:
What if we just declare the following:
```c++
 	struct ib_counter_element_t {
		MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE)
		Atomic_relaxed<Type> value;
	};
```

Reply:
done.

also, I tested the said approach against 10.5 and observed a regression of 5-10% with point-select and a gain 20% with read-only so I would not recommend this patch for 10.5

with 10.6 there is gain in point-select (5-7%) and ro (20-27%).

will post detailed results in the jira tickets.